### PR TITLE
Simplify IsErrorStatusMessage()

### PR DESCRIPTION
**Pull request by vineeth-thumma**
- Noticed that the retries are not working for Request canceled errors as the structure returned to the cosmos db client doesnt conform to Error struct expected by IsErrorStatusMessage
- Simplify IsErrorStatusMessage() to support any struct that implements error interface

### CODE REVIEW

File: pkg/gencosmosdb/cosmosdb/cosmosdb.go [hunk 1]
```diff
@@ -49,11 +49,7 @@ func IsErrorStatusMessage(err error, errorString string) bool {
 	if err == nil {
 		return false
 	}
-	if err2, ok := err.(*Error); ok {
-		return strings.Contains(err2.Message, errorString)
-	} else {
-		return strings.Contains(err.Error(), errorString)
-	}
+	return strings.Contains(err.Error(), errorString)
```
Comment (julienstroheker) [inline pkg/gencosmosdb/cosmosdb/cosmosdb.go:52]: The purpose of the function (and the name) does not make sense anymore no ? `IsErrorStatusMessage`

Also should we update UTs ?

Comment (vineeth-thumma) [reply pkg/gencosmosdb/cosmosdb/cosmosdb.go]: Thats a valid point, refactored it. UTs are updated as part of previous change here: https://github.com/julienstroheker/go-cosmosdb/pull/22 (removed the UTs that are not relevant anymore)

File: pkg/gencosmosdb/cosmosdb/cosmosdb.go [hunk 2]
```diff
@@ -77,7 +64,7 @@ func RetryOnHttpStatusOrError(f func() error, statusCode int, errorString ...str
 			var containsErrorString bool
 			if len(errorString) >= 1 {
 				for _, msg := range errorString {
-					if IsErrorStatusMessage(err, msg) {
+					if err != nil && strings.Contains(err.Error(), msg) {
```
Comment (julienstroheker) [inline pkg/gencosmosdb/cosmosdb/cosmosdb.go:67]: Just want to mention this is case sensitive : https://play.golang.com/p/ZVDc6Ay2V53

Not sure if this what you want ?

Comment (vineeth-thumma) [reply pkg/gencosmosdb/cosmosdb/cosmosdb.go]: Updated to make it case in-sensitive

**Review by julienstroheker — APPROVED** [review]

**Pull request merged**